src/photon.cpp

```diff
@@ -248,7 +248,7 @@ PhotonInteraction::PhotonInteraction(hid_t group)
   // Create Compton profile CDF
   n_profile_ = data::compton_profile_pz_size;
   profile_cdf_ = xt::empty<double>({n_shell, n_profile_});
-  for (int i = 0; i < n_shell; ++i) {
+  for (int i = 0; i < profile_pdf_.shape(0); ++i) {
```

Comment:
Note that I also added a bug fix in this block of code in openmc-dev/openmc#2297. Probably want to backport that one as well while you're at it.

Reply:
Thanks for letting me know about that one! I've added in the changes from that PR as well.